controller/workitem_blackbox_test.go

```diff
@@ -1659,6 +1660,170 @@ func (s *WorkItem2Suite) TestWI2ListByStateFilterOKModifiedUsingIfNoneMatchIfMod
 	require.Empty(s.T(), toBeFound, "failed to find these work items in list: %+v", toBeFound)
 }
 
+func (s *WorkItem2Suite) TestListOrder() {
+	s.T().Run("list with explicit order", func(t *testing.T) {
+		fxt := tf.NewTestFixture(t, s.DB,
+			tf.Iterations(2),
+			tf.WorkItems(10, func(fxt *tf.TestFixture, idx int) error {
+				switch idx {
+				case 0, 1, 2, 3, 4, 5, 6:
+					fxt.WorkItems[idx].Fields[workitem.SystemState] = "open"
+				default:
+					fxt.WorkItems[idx].Fields[workitem.SystemState] = "new"
+				}
+				return nil
+			}),
+		)
+
+		t.Run("by created descending", func(t *testing.T) {
+			// when
+			exp := ptr.String(`{"system.state": "open"}`)
+			sort := ptr.String("-created")
+			_, actualWIs := test.ListWorkitemsOK(s.T(), s.svc.Context, s.svc, s.workitemsCtrl, fxt.Spaces[0].ID, exp, nil, nil, nil, nil, nil, nil, nil, nil, nil, sort, nil, nil)
+			// then
+			require.NotNil(s.T(), actualWIs)
+			require.Len(s.T(), actualWIs.Data, 7)
+			toBeFound := id.Slice{
+				fxt.WorkItems[0].ID,
+				fxt.WorkItems[1].ID,
+				fxt.WorkItems[2].ID,
+				fxt.WorkItems[3].ID,
+				fxt.WorkItems[4].ID,
+				fxt.WorkItems[5].ID,
+				fxt.WorkItems[6].ID,
+			}.ToMap()
+
+			for i := 0; i <= 6; i++ {
+				require.Equal(t, fxt.WorkItems[i].ID, *actualWIs.Data[6-i].ID)
+			}
+
+			for _, wi := range actualWIs.Data {
+				_, ok := toBeFound[*wi.ID]
```

Comment:
@baijum here you are just checking that the workitem present in `actualWI` is present in `toBeFound` or not. There should be a proper check that the results in `actualWI` are sorted according to created descending i.e. the first item in actualWI has the latest time and last item has the oldest time.

Reply:
there should be a proper check of the createdAt values of the workitems present in `actualWI`

Reply:
This should be addressed for all the subtests for `TestListOrder`

Reply:
Order is already checked in the previous for loop.